src/hooks/useENSRegistrationStepHandler.tsx

```diff
@@ -166,16 +192,27 @@ export default function useENSRegistrationStepHandler(observer = true) {
           registrationParameters?.commitTransactionConfirmedAt ||
             msBlockTimestamp
         );
-        if (secs > ENS_SECONDS_WAIT) setReadyToRegister(true);
+        if (
+          (secs > ENS_SECONDS_WAIT_WITH_PADDING &&
+            secondsSinceCommitConfirmed > ENS_SECONDS_WAIT_WITH_PADDING) ||
+          // sometimes the provider.getBlock('latest) takes a long time to update to newest block
+          secondsSinceCommitConfirmed > ENS_SECONDS_WAIT_PROVIDER_PADDING
+        )
+          setReadyToRegister(true);
         // eslint-disable-next-line no-empty
       } catch (e) {}
     };
-    if (secondsSinceCommitConfirmed >= ENS_SECONDS_WAIT) {
+    if (
+      secondsSinceCommitConfirmed % 2 === 0 &&
+      secondsSinceCommitConfirmed >= ENS_SECONDS_WAIT &&
+      !readyToRegister
+    ) {
```

Comment:
again there's no need to check the block times if `!readyToRegister`, we were updating it all the time as the also changed with `secondsSinceCommitConfirmed` changes. And also let's check every 2 secs